### PR TITLE
distsql: set NodeID and FlowID in logging contexts

### PR DESCRIPTION
**Pull request by RaduBerinde**
Please ignore the first commit (r1), it is a separate PR.

Sample log output:

```
I160727 14:38:38.212306 sql/distsql/tablereader.go:233  [node: 2] [flow: 4fa99b27-9834-47e3-b58e-cfeb005b0f5b] [TableReader: 51] pushing row [83 11]
I160727 14:38:38.212335 sql/distsql/tablereader.go:233  [node: 2] [flow: 4fa99b27-9834-47e3-b58e-cfeb005b0f5b] [TableReader: 51] pushing row [92 11]
I160727 14:38:38.212345 sql/distsql/outbox.go:101  [node: 2] [flow: 4fa99b27-9834-47e3-b58e-cfeb005b0f5b] flushing outbox
I160727 14:38:38.212364 sql/distsql/outbox.go:114  [node: 2] [flow: 4fa99b27-9834-47e3-b58e-cfeb005b0f5b] outbox flushed
I160727 14:38:38.212377 sql/distsql/inbound.go:80  [node: 3] [flow: 4fa99b27-9834-47e3-b58e-cfeb005b0f5b] inbound stream pushing row [81 9]
```

@tschottdorf @jordanlewis

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/8066)

### CODE REVIEW

**Comment by jordanlewis** [comment]
Reviewed 22 of 22 files at r1.
Review status: 16 of 22 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/distsql/flow.go, line 85 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhgf7sWld9H2We5tUx:-KNhgf7sWld9H2We5tUy:1374822074) ([raw file](https://github.com/cockroachdb/cockroach/blob/fca4a6634824b239facb060c8bd526a4f2efdb2f/sql/distsql/flow.go#L85)):*
> ```Go
> 	simpleFlowConsumer RowReceiver,
> ) *Flow {
> 	// TODO(radu): UUIDs are pretty long, perhaps an int64 would be sufficient
> ```

Agreed that UUIDs are a bit long, especially for outputting in the logs. Maybe we could print a short prefix instead if we'd like to keep the UUID internally.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066)*

**Comment by petermattis** [comment]
Review status: 16 of 22 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/distsql/flow.go, line 85 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhgf7sWld9H2We5tUx:-KNhhHglUfjox6Vy1Hp_:587908693) ([raw file](https://github.com/cockroachdb/cockroach/blob/fca4a6634824b239facb060c8bd526a4f2efdb2f/sql/distsql/flow.go#L85)):*
<details><summary><i>Previously, jordanlewis (Jordan Lewis) wrote…</i></summary>
> Agreed that UUIDs are a bit long, especially for outputting in the logs. Maybe we could print a short prefix instead if we'd like to keep the UUID internally.
</details>
`util/uuid.UUID.Short()`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066)*

**Comment by RaduBerinde** [comment]
Current diff r1->r3.

---

Review status: 16 of 22 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/distsql/flow.go, line 85 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhgf7sWld9H2We5tUx:-KNhl-apZogans79g1Xh:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/fca4a6634824b239facb060c8bd526a4f2efdb2f/sql/distsql/flow.go#L85)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> `util/uuid.UUID.Short()`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KNhl03sEfeMgxi0zspP:-29048018)*

**Comment by tamird** [comment]
Reviewed 22 of 22 files at r1, 5 of 6 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/distsql/flow.go, line 85 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhgf7sWld9H2We5tUx:-KNhpxSftExPWMjAvSUb:2119796242) ([raw file](https://github.com/cockroachdb/cockroach/blob/fca4a6634824b239facb060c8bd526a4f2efdb2f/sql/distsql/flow.go#L85)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary>
> Done.
</details>
probably good to remove this TODO now.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066)*

**Comment by RaduBerinde** [comment]
Current diff r1->r4.

---

Review status: 21 of 22 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[sql/distsql/flow.go, line 85 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhgf7sWld9H2We5tUx:-KNhyi8QFLVTABj3pW8Y:1597093114) ([raw file](https://github.com/cockroachdb/cockroach/blob/fca4a6634824b239facb060c8bd526a4f2efdb2f/sql/distsql/flow.go#L85)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> probably good to remove this TODO now.
</details>
Ah, duh, thanks.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KNhypA-LFIJpu6Id4IY:-29047987)*

**Comment by RaduBerinde** [comment]
Updated. I recommend looking at the diff in github as Reviewable is thoroughly confused. @andreimatei 

---

Review status: 3 of 8 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KOGqPW24Y84FNRr-hkN:43079080)*

**Comment by andreimatei** [comment]
Could we get the context info in front of the file/line num in the logs?

**Comment by RaduBerinde** [comment]
It would require a bit of dancing (we trace the message without the file/line) but I don't see why not. @petermattis what do you think?

Either way it would be a separate PR, this PR isn't related to the general format of log messages.

---

Review status: 3 of 8 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KOH2UELPn4AVmnx4Sf7:-390243071)*

File: sql/distsql/joinreader.go [hunk 1]
```diff
@@ -95,10 +93,10 @@ func (jr *joinReader) mainLoop() error {
 	var alloc sqlbase.DatumAlloc
 	spans := make(sqlbase.Spans, 0, joinReaderBatchSize)
 
-	// TODO(radu): add info about the joinreader in the context.
+	ctx := log.WithLogTagInt(jr.ctx, "JoinReader", int(jr.desc.ID))
```
Comment (andreimatei) [inline sql/distsql/joinreader.go]: can this move to the ctor?

File: server/server.go [hunk 2]
```diff
@@ -421,6 +420,7 @@ func (s *Server) Start() error {
 	s.node.startWriteSummaries(s.ctx.MetricsSampleInterval)
 
 	s.sqlExecutor.SetNodeID(s.node.Descriptor.NodeID)
```
Comment (andreimatei) [inline server/server.go]: I forgot how this works; when does is this `s.node.NodeID` assigned? This seems like a late place to init these fields on the executor and distsqlServer.

**Comment by andreimatei** [comment]
LGTM

**Comment by RaduBerinde** [comment]
TFTR!

---

Review status: 0 of 9 files reviewed at latest revision, 3 unresolved discussions, some commit checks pending.

---

*[server/server.go, line 422 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhg66J0R41iRBEILdG-r5-422:-KOH5Vm_gVJtrqgzRLgK:-1772191250) ([raw file](https://github.com/cockroachdb/cockroach/blob/c07b5c3cc95d142f71f23b1857a35afbca4e14b0/server/server.go#L422)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary>
> I forgot how this works; when does is this `s.node.NodeID` assigned? This seems like a late place to init these fields on the executor and distsqlServer.
</details>
Unfortunately it can be as late as the `node.start()` call above (`start->initStores->initNodeID`).

---

*[sql/distsql/joinreader.go, line 96 \[r5\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-KNhg66J0R41iRBEILdN-r5-96:-KOH5jGSGKTdd1sYEadF:1972804061) ([raw file](https://github.com/cockroachdb/cockroach/blob/c07b5c3cc95d142f71f23b1857a35afbca4e14b0/sql/distsql/joinreader.go#L96)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary>
> can this move to the ctor?
</details>
Done. Also renamed some `ctx *FlowCtx`s to `flowCtx`, and changed the embedded `Context` in `FlowCtx` and `ServerContext` to regular members to avoid confusion.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KOH7myiRpBaC5ABsTXY:-485616692)*

**Comment by andreimatei** [comment]
LGTM

**Comment by petermattis** [comment]
Putting the context tags before the file/line number is doable, though mildly irritating. I don't see a huge benefit to doing so. More interesting would be to include the goroutine ID. See https://github.com/petermattis/goid.

---

Review status: 0 of 9 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KOHE-YeKk7MhlQiL1GF:-9597875)*

**Comment by RaduBerinde** [comment]
I like the `goid` stuff! I know that the go folks are dead set against providing an API to get the current goroutine; are we reasonably confident we will always be able to have a working hack for future go versions?

---

Review status: 0 of 9 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/8066#-:-KOHJqRTKSxe1sEreUPN:208303033)*

**Pull request merged**